File: CS50X/Week 6/sentimental-cash/cash.py
from cs50 import get_float


def main():
    # Prompt user for the amount of change owed
    while True:
        change = get_float("Change owed: ")
        if change >= 0:
            break

    # Convert dollars to cents to avoid floating-point imprecision
    cents = round(change * 100)

    # Initialize the count of coins
    coins = 0

    # Calculate the number of quarters
    coins += cents // 25
    cents %= 25

    # Calculate the number of dimes
    coins += cents // 10
    cents %= 10

    # Calculate the number of nickels
    coins += cents // 5
    cents %= 5

    # Calculate the number of pennies
    coins += cents

    # Output the total number of coins
    print(coins)


if __name__ == "__main__":
    main()

File: CS50X/Week 1/cash/cash.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int change;
    // Define coin values
    int quarter = 25;
    int dime = 10;
    int nickel = 5;
    int penny = 1;
    // Calculate the minimum coins needed
    int coins = 0;

    // Prompt user for input until a valid positive integer is given
    do
    {
        change = get_int("Change: ");
    }
    while (change < 1);

    // Calculate the minimum coins needed
    coins += change / quarter;
    // Remaining change after using quarters
    change %= quarter;

    // Number of dimes
    coins += change / dime;
    // Remaining change after using dimes
    change %= dime;

    // Number of nickels
    coins += change / nickel;
    // Remaining change after using nickels
    change %= nickel;

    // Number of pennies
    coins += change;

    printf("%d\n", coins);
}
